src/cmd/getM365/onedrive/get_item.go

```diff
@@ -37,132 +36,119 @@ var (
-	fs.StringVar(&m365ID, "m365ID", "", "m365 identifier for object")
+	fs.StringVar(&m365ID, "id", "", "m365 identifier for object")
 	fs.StringVar(&user, "user", "", "m365 user id of M365 user")
 	fs.StringVar(&tenant, "tenant", "", "m365 identifier for the tenant")
 
 	cobra.CheckErr(exCmd.MarkPersistentFlagRequired("user"))
-	cobra.CheckErr(exCmd.MarkPersistentFlagRequired("tenant"))
-	cobra.CheckErr(exCmd.MarkPersistentFlagRequired("m365ID"))
+	cobra.CheckErr(exCmd.MarkPersistentFlagRequired("id"))
 
 	parent.AddCommand(exCmd)
 }
 
 func handleOneDriveCmd(cmd *cobra.Command, args []string) error {
-	ctx := cmd.Context()
-
 	if utils.HasNoFlagsAndShownHelp(cmd) {
 		return nil
 	}
 
-	gc, creds, err := gutils.GetGC(ctx, tenant)
+	tid := common.First(tenant, os.Getenv(account.AzureTenantID))
+
+	ctx := clues.Add(
+		cmd.Context(),
+		"item_id", m365ID,
+		"resource_owner", user,
+		"tenant", tid)
+
+	// get account info
+	creds := account.M365Config{
+		M365:          credentials.GetM365(),
+		AzureTenantID: tid,
+	}
+
+	// todo: swap to drive api client, when finished.
+	adpt, err := graph.CreateAdapter(tid, creds.AzureClientID, creds.AzureClientSecret)
 	if err != nil {
-		return err
+		return Only(ctx, clues.Wrap(err, "creating graph adapter"))
 	}
 
-	err = runDisplayM365JSON(ctx, gc, creds, user, m365ID)
+	err = runDisplayM365JSON(ctx, graph.NewService(adpt), creds, user, m365ID)
 	if err != nil {
 		cmd.SilenceUsage = true
 		cmd.SilenceErrors = true
 
-		return errors.Wrapf(err, "unable to fetch from M365: %s", m365ID)
+		return Only(ctx, clues.Wrap(err, "getting item"))
 	}
 
 	return nil
 }
 
 type itemData struct {
-	Size int `json:"size"`
+	Size int64 `json:"size"`
+}
+
+type itemPrintable struct {
+	Info        string   `json:"info"`
+	Permissions string   `json:"permissions"`
+	Data        itemData `json:"data"`
 }
 
-type item struct {
-	Info        models.DriveItemable
-	Permissions models.PermissionCollectionResponseable
-	Data        itemData
+func (i itemPrintable) MinimumPrintable() any {
+	return i
 }
 
 func runDisplayM365JSON(
 	ctx context.Context,
-	gc *connector.GraphConnector,
+	srv graph.Servicer,
 	creds account.M365Config,
 	user, itemID string,
 ) error {
-	driveID, err := getDriveID(ctx, gc.Service, user)
+	drive, err := api.GetDriveByID(ctx, srv, user)
 	if err != nil {
 		return err
 	}
 
-	it := item{}
+	driveID := ptr.Val(drive.GetId())
 
-	item, err := api.GetDriveItem(ctx, gc.Service, driveID, itemID)
+	it := itemPrintable{}
+
+	item, err := api.GetDriveItem(ctx, srv, driveID, itemID)
 	if err != nil {
 		return err
 	}
 
-	it.Info = item
+	it.Data.Size = ptr.Val(item.GetSize())
 
-	perms, err := api.GetItemPermission(ctx, gc.Service, driveID, itemID)
+	sInfo, err := serializeObject(item)
 	if err != nil {
 		return err
 	}
 
-	it.Permissions = perms
-
-	if item != nil {
-		content, err := getDriveItemContent(item)
-		if err != nil {
-			return err
-		}
-
-		it.Data.Size = len(content)
-	}
```

Comment:
While we can get size from `item.GetSize()`, this call was to ensure that we are able to download the file. I'm adding this back.